test/common/run-tests

```diff
@@ -301,139 +315,135 @@ def detect_tests(test_files, image, opts):
     return (serial_tests, parallel_tests)
 
 
-def run(opts, image):
-    result = 0
-    jobs = 1 if opts.list else opts.jobs
-    start_time = time.time()
+def list_tests(opts):
+    test_files = glob.glob(os.path.join(opts.test_dir, "check-*"))
+    (serial_tests, parallel_tests) = detect_tests(test_files, "dummy", opts)
+    names = [t.command[-1] for t in serial_tests] + [t.command[-1] for t in parallel_tests]
+    names.sort()
+    for n in names:
```

Comment:
`for n in sorted(names):`
is nice

Reply:
FTR: I actually thought in-place sorting of an list was more efficient than sorting an iterator; but it hardly matters here.